### PR TITLE
HDT format added and turtle sniffer improved

### PR DESCRIPTION
**Pull request by jjkoehorst**
We have added support for HDT binary RDF files and improved the TURTLE sniffer.

### CODE REVIEW

File: lib/galaxy/datatypes/triples.py [hunk 1]
```diff
@@ -156,3 +160,28 @@ def set_peek( self, dataset, is_multi_byte=False ):
...
+class HDT( binary.Binary, Triples ):
+    """
+    The HDT triple data format
+    """
+    edam_format = "format_2376"
+    file_ext = "hdt"
+
+    def sniff( self, filename ):
+        with open(filename, "rb") as f:
+            if f.readline( 1024 ).startswith("$HDT"):
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/triples.py]: `f.read(4) == "$HDT"` instead of `f.readline( 1024 ).startswith("$HDT")` should be enough here?

Comment (jjkoehorst) [inline lib/galaxy/datatypes/triples.py]: Yes indeed that should be sufficient

**Comment by nsoranzo** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
:+1:

**Comment by bgruening** [comment]
@jjkoehorst nice addition. Thanks a lot!

**Pull request merged**